src/main/kotlin/nl/info/zac/documentcreation/DocumentCreationService.kt

```diff
@@ -5,128 +5,54 @@
-        documentCreationDataAttended: DocumentCreationDataAttended
-    ): DocumentCreationAttendedResponse =
-        documentCreationDataConverter.createData(
-            loggedInUser = loggedInUserInstance.get(),
-            zaak = documentCreationDataAttended.zaak,
-            taskId = documentCreationDataAttended.taskId
-        ).runCatching {
-            createSmartDocumentForAttendedFlow(documentCreationDataAttended).let {
-                smartDocumentsService.createDocumentAttended(
-                    data = this,
-                    smartDocument = it
-                )
-            }
-        }.onFailure {
-            LOG.log(Level.WARNING, "Failed to create document with attended flow", it)
-        }.getOrThrow()
-
     /**
-     * Creates a SmartDocument object for the attended flow.
-     * In this flow the description of the zaaktype of the zaak in the provided document creation data is used
-     * as the SmartDocuments template group.
-     */
-    private fun createSmartDocumentForAttendedFlow(creationDataUnattended: DocumentCreationDataAttended) =
-        SmartDocument(
-            selection = Selection(
-                templateGroup = smartDocumentsTemplatesService.getTemplateGroupName(
-                    creationDataUnattended.templateGroupId
-                ),
-                template = smartDocumentsTemplatesService.getTemplateName(creationDataUnattended.templateId)
-            ),
-            variables = Variables(
-                // SmartDocuments use file extensions (without the leading `.`) instead of media types
-                // as the output format
-                outputFormats = listOf(
-                    OutputFormat(MediaTypes.Application.MS_WORD_OPEN_XML.extensions.first().drop(1))
-                ),
-                redirectMethod = REDIRECT_METHOD,
-                redirectUrl = documentCreationCallbackUrl(
-                    creationDataUnattended.zaak.uuid,
-                    creationDataUnattended.taskId,
-                    creationDataUnattended.templateGroupId,
-                    creationDataUnattended.templateId,
-                    creationDataUnattended.title,
-                    creationDataUnattended.description,
-                    creationDataUnattended.creationDate,
-                    creationDataUnattended.author ?: loggedInUserInstance.get().getFullName()
-                ).toString()
-            )
-        )
-
-    /**
-     * Download generated SmartDocuments file and store it as Informatieobject
+     * Download a generated SmartDocuments file and store it as an Informatieobject
      */
     fun storeDocument(
         zaak: Zaak,
         taskId: String? = null,
         fileId: String,
-        templateGroupId: String,
-        templateId: String,
         title: String,
         description: String?,
+        informatieobjecttypeUuid: UUID,
         creationDate: ZonedDateTime,
-        userName: String
+        userName: String,
```

Comment:
This dangling `,` does not compile does it?

Reply:
It actually does compile and detekt does not mind it either. Still it's a bit strange indeed

Reply:
Removed